aiida/cmdline/utils/migration/__init__.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Migration export files from old export versions to the newest, used by `verdi export migrate` command."""
+# pylint: disable=wildcard-import
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+from aiida.cmdline.utils import echo
+from aiida.common.exceptions import DanglingLinkError
+
+from .utils import *
```

Comment:
this is potentially dangerous and can make tracking problems difficult.

is there not an easier way to achieve this, e.g. by importing the respective functions using `importlib` when needed?

Reply:
I am copying the way it is done in other `__init__.py` files throughout AiiDA core (check e.g. [here](https://github.com/aiidateam/aiida_core/blob/develop/aiida/cmdline/__init__.py).

It would be good to know of an "AiiDA standard" for these things in internal sub-modules (which is what I would call this "utility").

Reply:
Or do you mean that I should specify an `__all__` in `utils.py` as well perhaps?

Reply:
this would also allow to move some of the helper function definitions inside  `v03_to_v04` to the top level without needing to be worried about name conflicts.

Reply:
> is there not a better way to achieve this, e.g. by having a "migrate" function in all of those modules and importing the respective function using `importlib.import_module` when needed?

I think I would need a more detailed explanation of this idea, sorry. I am not completely following what you intend the change should be.

Reply:
> I am copying the way it is done in other __init__.py files throughout AiiDA core (check e.g. here.

I see.

In the aiida-core example you gave, the purpose of `import *` is to make the corresponding names (which are many and are constantly growing) available to the user at a certain point of the module hierarchy. Except for this, the `__init__.py` does nothing.

In the PR here, your `__init__.py` actually does work and the functions it imports are needed to do this work.
Unless the purpose is to make the various migrate functions available to users, I think there is no good reason to use `import *` here.

Note: Instead of my suggestion with `importlib`, you can also simply hardcode the imports (which are unlikely to change over time).

Reply:
Okay! :)
Thank you for the clarification. I feel more confident in making the changes now.

I will take you up on your note and simply hardcode the functions that are needed.

Reply:
Done